cogs/make_applicant.py

```diff
@@ -0,0 +1,203 @@
+"""Contains cog classes for making a user into an applicant."""
+
+from collections.abc import Sequence
+
+__all__: Sequence[str] = (
+    "BaseMakeApplicantCog",
+    "MakeApplicantSlashCommandCog",
+    "MakeApplicantContextCommandsCog",
+)
+
+
+import logging
+from logging import Logger
+from typing import Final
+
+import discord
+
+from exceptions.does_not_exist import ApplicantRoleDoesNotExistError, GuildDoesNotExistError
+from utils import CommandChecks, TeXBotApplicationContext, TeXBotBaseCog
+
+logger: Logger = logging.getLogger("TeX-Bot")
+
+
+class BaseMakeApplicantCog(TeXBotBaseCog):
+    """
+    Base making-applicant cog container class.
+
+    Defines the methods for making users into group-applicants, that are called by
+    child cog container classes.
+    """
+
+    async def _perform_make_applicant(self, ctx: TeXBotApplicationContext, applicant_member: discord.Member) -> None:  # noqa: E501
+        """Perform the actual process of making the user into a group-applicant."""
+        main_guild: discord.Guild = ctx.bot.main_guild
+        applicant_role: discord.Role = await ctx.bot.applicant_role
+        guest_role: discord.Role = await ctx.bot.guest_role
+
+        intro_channel: discord.TextChannel | None = discord.utils.get(
+            main_guild.text_channels,
+            name="introductions",
+        )
+
+        initial_response: discord.Interaction | discord.WebhookMessage = await ctx.respond(
+            ":hourglass: Attempting to make user an applicant... :hourglass:",
+            ephemeral=True,
+        )
+
+        if applicant_member.bot:
+            await initial_response.edit(content="Cannot make a bot user an applicant!")
+            await self.command_send_error(ctx, message="Cannot make a bot user an applicant!")
+            return
```

Comment:
Place this block above the deferral response (onto line 42) and remove call to `await initial_response.edit(...` to prevent duplicate error message being sent twice.